susemanager-utils/susemanager-sls/salt/util/mgr_switch_to_venv_minion.sls

```diff
@@ -0,0 +1,100 @@
+{%- set susemanager_conf='/etc/salt/minion.d/susemanager.conf' %}
+{%- set venv_susemanager_conf='/etc/opt/venv-salt-minion/minion.d/susemanager.conf' %}
+{%- set managed_minion=salt['file.file_exists'](susemanager_conf) and
+                       not salt['file.replace'](susemanager_conf, '^master: .*', 'master: ' + pillar['mgr_server'],
+                                                dry_run=True, show_changes=False, ignore_if_missing=True) %}
+{%- set venv_managed_minion=salt['file.file_exists'](venv_susemanager_conf) and
+                            not salt['file.replace'](venv_susemanager_conf, '^master: .*', 'master: ' + pillar['mgr_server'],
+                                                     dry_run=True, show_changes=False, ignore_if_missing=True) %}
+{%- if managed_minion or venv_managed_minion %}
+{%- set pkgs_installed = salt['pkg.info_installed']() %}
+{%- set venv_minion_installed = pkgs_installed.get('venv-salt-minion', False) and True %}
+{%- set venv_minion_available = venv_minion_installed or salt['pkg.latest_version']('venv-salt-minion') or False %}
+{%- if venv_minion_available %}
+venv-salt-minion-pkg:
+  pkg.installed:
+    - name: venv-salt-minion
+    - onlyif:
+      - ([ {{ venv_minion_installed }} = "False" ])
+
+copy-salt-minion-id:
+  file.copy:
+    - name: /etc/opt/venv-salt-minion/minion_id
+    - source: /etc/salt/minion_id
+    - require:
+      - pkg: venv-salt-minion-pkg
+    - onlyif:
+      - test -f /etc/salt/minion_id
+
+copy-salt-minion-configs:
+  cmd.run:
+    - name: cp -r /etc/salt/minion.d /etc/opt/venv-salt-minion/
+    - require:
+      - pkg: venv-salt-minion-pkg
+    - onlyif:
+      - ([ {{ venv_managed_minion }} = "False" ])
+
+copy-salt-minion-keys:
+  cmd.run:
+    - name: cp -r /etc/salt/pki/minion/minion* /etc/opt/venv-salt-minion/pki/minion/
```

Comment:
Are we sure that the destination directory exists? For the normal salt package it does not:
```
$> rpm -qf /etc/salt/pki/minion/
file /etc/salt/pki/minion is not owned by any package
```

Reply:
On my test systems it works for openSUSE/CentOS and Raspbian, it's possible that I've appended the dir itself to the package, but will check it anyway.

Reply:
In the original salt package the minion directory is not part of the package. This can be different in the new venv package.
If this is the case, fine. Otherwise I would suggest to set the makedirs option.

Reply:
`%dir %{_sysconfdir}/opt/%{venv}/pki/minion/` is in the `%files` section of the spec and it's reflected to `deb` with `debbuild` also. I beleive we are safe with it.